sunbeam-python/sunbeam/features/observability/etc/deploy-grafana-agent/variables.tf

```diff
@@ -17,6 +17,7 @@
 
 variable "principal-application" {
   description = "Name of the deployed principal application that integrates with grafana-agent"
+  default = ""
```

Comment:
Can we use null instead of empty string (to use similar pattern with other variables)

Reply:
I am thinking if we can make `principal-application` as a list of applications grafana-agent have to be integrated with. (variable name should be renamed)

This will simplify the terraform plan to have single resource definition to openstack-hypervisor, k8s.
Also avoids hardcoding of application name `k8s` in the terraform plan.